src/geometry.rs

```diff
@@ -1141,4 +1348,106 @@ mod tests {
 
         assert!(!geometry_with_orphans.has_no_orphan_normals());
     }
+    #[test]
+    fn test_remove_orphan_vertices() {
+        let (faces, vertices, _normals) = quad_with_normals();
+        let extra_vertex = vec![v(0.0, 0.0, 0.0, [0.0, 0.0, 0.0], 1.0)];
+        let vertices_extended = [&extra_vertex[..], &vertices[..]].concat();
+        let faces_renumbered_to_match_extend_vertices: Vec<_> = faces
+            .iter()
+            .map(|f| (f.vertices.0 + 1, f.vertices.1 + 1, f.vertices.2 + 1))
+            .collect();
+
+        let faces_length = &faces.len();
+
+        let (faces_purged, vertices_purged) =
+            remove_orphan_vertices(faces_renumbered_to_match_extend_vertices, vertices_extended);
+
+        let faces_purged_length = &faces_purged.len();
+
+        assert_eq!(faces_length, faces_purged_length);
+        assert_eq!(vertices_purged, vertices);
+    }
+
+    #[test]
+    fn test_remove_orphan_normals() {
+        let (faces, _vertices, normals) = quad_with_normals();
+        let extra_normal = vec![n(0.0, 0.0, 0.0)];
+        let normals_extended = [&extra_normal[..], &normals[..]].concat();
+
+        let faces_renumbered_to_match_extend_normals: Vec<_> = faces
+            .iter()
+            .map(|f| {
+                TriangleFace::new_separate(
+                    f.vertices.0,
+                    f.vertices.1,
+                    f.vertices.2,
+                    f.normals.0 + 1,
+                    f.normals.1 + 1,
+                    f.normals.2 + 1,
+                )
+            })
+            .collect();
+
+        let faces_length = &faces.len();
+
+        let (faces_purged, normals_purged) =
+            remove_orphan_normals(faces_renumbered_to_match_extend_normals, normals_extended);
+
+        let faces_purged_length = &faces_purged.len();
+
+        assert_eq!(faces_length, faces_purged_length);
+        assert_eq!(normals_purged, normals);
+    }
+
+    #[test]
+    fn test_geometry_from_triangle_faces_with_vertices_and_computed_normals_remove_orphans() {
+        let (faces, vertices, _normals) = quad_with_normals();
+        let extra_vertex = vec![v(0.0, 0.0, 0.0, [0.0, 0.0, 0.0], 1.0)];
+        let vertices_extended = [&extra_vertex[..], &vertices[..]].concat();
+        let faces_renumbered_to_match_extend_vertices: Vec<_> = faces
```

Comment:
Same as above. Why prepend and renumber when you can append?

Reply:
same as above